packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import uniqueid from 'lodash.uniqueid';
+import classNames from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+    * Whether or not the Switch is enabled ("ON").
+    */
+  isOn: PropTypes.bool,
+
+  /**
+    * If true, the user won't be able to toggle the switch.
+    */
+  isDisabled: PropTypes.bool,
+
+  /**
+    * Text of the label.
+    */
+  labelText: PropTypes.node.isRequired,
+
+  /**
+    * Callback fired when the state is changed. Parameters are (event).
+    */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  isOn: false,
+  isDisabled: false,
+  onChange: undefined,
+};
+
+const Switch = ({
+  isOn,
+  isDisabled,
+  onChange,
+  labelText,
+  ...customProps
+}) => {
+  const [focused, setFocus] = useState(false);
+  const sliderButtonRef = useRef(null);
+
+  const handleClick = (event) => {
+    setFocus(true);
+    sliderButtonRef.current.focus();
+    onChange(event);
```

Comment:
This doesn't do a guard check against an undefined onChange event. The onChange event either needs to be required or this should be wrapped in an if check. 

```jsx
if (onChange) {
  onChange(event)
}
```

Reply:
I think adding condition for Onchange seems to be more precise as of now rather than making onChange as required until we really need it to be required for Switch Toggle.
Added condition [here](https://github.com/cerner/terra-core/blob/895c4fad7bb180e27e1919296ac90eeaa805dea1/packages/terra-switch/src/Switch.jsx#L75)